datadog/resource_datadog_logs_pipeline.go

```diff
@@ -0,0 +1,705 @@
+				ddCategory.SetFilter(buildDatadogFilter(tfFilter[0].(map[string]interface{})))
+			}
+
+			ddCategories[i] = ddCategory
+		}
+		ddCategory.Categories = ddCategories
+	}
+	return ddCategory
+}
+
+func buildDatadogAttributeRemapper(tfProcessor map[string]interface{}) datadog.AttributeRemapper {
+	ddAttribute := datadog.AttributeRemapper{}
+	if ddSources := buildDatadogSources(tfProcessor); ddSources != nil {
+		ddAttribute.Sources = ddSources
+	}
+	if tfSourceType, exists := tfProcessor["source_type"].(string); exists {
+		ddAttribute.SetSourceType(tfSourceType)
+	}
+	if tfTarget, exists := tfProcessor["target"].(string); exists {
+		ddAttribute.SetTarget(tfTarget)
+	}
+	if tfTargetType, exists := tfProcessor["target_type"].(string); exists {
+		ddAttribute.SetTargetType(tfTargetType)
+	}
+	if tfPreserveSource, exists := tfProcessor["preserve_source"].(bool); exists {
+		ddAttribute.SetPreserveSource(tfPreserveSource)
+	}
+	if tfOverrideOnConflict, exists := tfProcessor["override_on_conflict"].(bool); exists {
+		ddAttribute.SetOverrideOnConflict(tfOverrideOnConflict)
+	}
+	return ddAttribute
+}
+
+func buildDatadogSources(tfProcessor map[string]interface{}) []string {
+	if tfSources, exists := tfProcessor["sources"].([]interface{}); exists && len(tfSources) > 0 {
+		ddSources := make([]string, len(tfSources))
+		for i, tfSource := range tfSources {
+			ddSources[i] = tfSource.(string)
+		}
+		return ddSources
+	}
+	return nil
+}
+
+func buildDatadogArithmeticProcessor(tfProcessor map[string]interface{}) datadog.ArithmeticProcessor {
+	ddArithmetic := datadog.ArithmeticProcessor{}
+	if tfTarget, exists := tfProcessor["target"].(string); exists {
+		ddArithmetic.SetTarget(tfTarget)
+	}
+	if tfExpression, exists := tfProcessor["expression"].(string); exists {
+		ddArithmetic.SetExpression(tfExpression)
+	}
+	if tfIsReplaceMissing, exists := tfProcessor["is_replace_missing"].(bool); exists {
+		ddArithmetic.SetIsReplaceMissing(tfIsReplaceMissing)
+	}
+	return ddArithmetic
+}
+
+func buildDatadogFilter(tfFilter map[string]interface{}) datadog.FilterConfiguration {
+	ddFilter := datadog.FilterConfiguration{}
+	if tfQuery, exists := tfFilter["query"].(string); exists {
+		ddFilter.SetQuery(tfQuery)
+	}
+	return ddFilter
+}
+
+func getPipelineSchema(isNested bool) map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name":       {Type: schema.TypeString, Required: true},
+		"is_enabled": {Type: schema.TypeBool, Optional: true},
+		"filter": {
+			Type:     schema.TypeList,
+			Required: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"query": {Type: schema.TypeString, Required: true},
+				},
+			},
+		},
+		"processor": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: getProcessorSchema(isNested),
+			},
+		},
+	}
+}
+
+func getProcessorSchema(isNested bool) map[string]*schema.Schema {
```

Comment:
I hope I'm not missing something, but it feels like you could make this function much simpler if you had a map such as `{tfArithmeticProcessor: arithmeticProcessor, ...}` and then just loop over it and add it's elements to `processorsSchema`. Does that make sense?